docs/types/standard-linter-v2.md

```diff
@@ -0,0 +1,46 @@
+# Standard Linter v2
+
+This document describes the type of Standard Linter v2, for an example in how to implement this in your own package see the see the [Example Usage](../examples/standard-linter-v2.md) document.
+
+## Type
+
+```js
+type Linter = {
+  // Automatically added, not guaranteed
+  __$sb_linter_version: 2,
+  __$sb_linter_activated: boolean,
+  __$sb_linter_request_latest: number,
+  __$sb_linter_request_last_received: number,
+
+  // From providers
+  name: string,
+  scope: 'file' | 'project',
+  lintOnFly: boolean,
+  grammarScopes: Array<string>,
+  lint(textEditor: TextEditor): ?Array<Message | MessageLegacy> | Promise<?Array<Message | MessageLegacy>>,
+}
+```
+
+## FAQs
+
+**Q**: What's the difference between `file` and `project` scopes?
+
+**A**: Results of `file` scoped linters are for individual files, therefore their result for a file is discarded as soon as it's TextEditor is closed. Their newer results only replace their previous results for that particular file. Examples of such linters are `ESLint`.
```

Comment:
"Their newer results only replace their previous results for that particular file."

I'm not sure this line should be in here, it's irrelevant for a standard Linter provider as it's just an internal part of how Linter is implemented to them.

If you want to keep it the wording is a bit awkward.